### ISSUE TITLE
"not authorized to perform: batch:SubmitJob on resource"

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaCloud/AWS.jl#617]
I'm more than likely doing something obviously wrong here, but I can't figure it out.

I'm trying to submit a batch job from an EC2 instance. It works fine from the AWS-CLI, using the role I assigned to the instance.

```
# AWS CLI call:
run(Cmd(["aws", "batch", "submit-job",
     "--job-name", "<job name>",
     "--job-definition", "<job definition name>:<version>",
     "--job-queue", "<job queue name>",
     "--parameters", "<parameters>",
     "--container-overrides", "resourceRequirements=[{value=$required_cores,type=VCPU},{value=$required_mem,type=MEMORY}]",
     "--timeout", "attemptDurationSeconds=$timeout"]))
# this works

# AWS.jl call:
using AWS
@service Batch
x = Batch.submit_job(
    "<job definition name>:<version>", 
    "<job name>",
    "<job queue name>",
    Dict(
        ["parameters" => "<parameters>",
        "containerOverrides" => "resourceRequirements=[{value=$required_cores,type=VCPU},{value=$required_mem,type=MEMORY}]",
        "timeout" => timeout]
    )
)

ERROR: AWS.AWSExceptions.AWSException: 403 -- User: arn:aws:sts::<iam number>:assumed-role/<role name>/<instance id> is not authorized to perform: batch:SubmitJob on resource: arn:aws:batch:us-east-2:<iam number>:job-definition/
```

One thing that stands out is that the resource shown in the error isn't fully qualified with the specific job definition. Is the API endpoint doing something different than the CLI by design? Or is the AWS.jl generating a wrong request? How can I find out?

**Comment by BioTurboNick** [comment]
Hmm...

If I do this:
```
Batch.submit_job(
    "<job definition name>:<version>", 
    "<job name>",
    "<job queue name>")
```

It accepts the job definition and job queue. If I put an incorrect definition or queue, I get an access denied that makes sense, citing the specific definition or queue I attempted to call.

But if I include the Dict for the other parameters, it seems to act like I had passed an empty string for the job definition and queue.
